final/tool/block_add.cpp

```diff
@@ -0,0 +1,19 @@
+#include "stdafx.h"
+#include "block_add.h"
+#include "block.h"
+#include "self_registered_in_factory.h"
+
+#pragma region add
+add::add(const std::vector<double>& parameters) : block(GetFactoryName(), 1, parameters)
+{
+  initial_value_ = parameters[0];
+}
+
+double add::calc(double input)
+{
+  is_blocktype_registered_;
+  return initial_value_ + input;
+};
+
+std::string self_registered_in_factory<add>::description_ = "Summarize input with parameter value. Example: add 5";

```

Comment:
mozda dodati konstruktor u 

self_registered_in_factory(const char* description, const char* name) pa ne treba svaki blok definirati svoje metode?
disklejmer: ovaj prijedlog podrazumijeva da su uvijek inicijalizirani u compile-timeu https://stackoverflow.com/questions/2001286/const-char-s-in-c

------
inicijalizaciju statica je moguce izbjeci tako da svaka klasa dogovorno ima const char* member description_ pa iskoristiti CRTP https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern
static std::string self_registered_in_factory::GetDescription() { return TBlock::description_; }

Reply:
first part not accepted.
the second part is partially accepted.